src/fold.rs

```diff
@@ -418,21 +434,29 @@ fn cast(expr: Expr, ctype: &Type) -> CompileResult<ExprType> {
 /// and we can use that to store the new type
 fn const_cast(token: &Literal, ctype: &Type) -> Option<Literal> {
```

Comment:
Can you add a test for this function? Covering the old parts of it seem easy enough too.

Reply:
While writing test cases, I found that `!'0'` is incorrectly parsed as `'0' != 0` instead of `(int)'0' != 0`. I don't have time to fix it tonight, but I'll push the test cases I have in the meantime.

Reply:
Are you going to do that in a separate PR?

Reply:
Got it! It wasn't a bug in `parse` at all, I wasn't folding casts to `char` properly.